### PR TITLE
Server commands

### PR DESCRIPTION
**Pull request by JasonLAGOUTE**
## **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [X] Docs have been added / updated (for bug fixes / features)

## **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
This PR introduces the following:
- Final server of zappy
- all commands for graphic protocol
- all commands for ai protocol

## **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
The binary needs command line arguments to correctly execute.

### CODE REVIEW

**Review by noerio — APPROVED** [review]
Seems good

**Comment by Sinan-Karakaya** [comment]
@JasonLAGOUTE @thomasjuin1 Is this PR ready to merge?

File: zappy_server/include/map.h [hunk 1]
```diff
@@ -23,16 +24,35 @@
...
+        tiles_t **tiles;
+        size_t *density[LENGTH_DENSITY];
+    } map_t;
+
+    typedef struct vector_s {
+        int x;
+        int y;
+    } vector_t;
+
+    map_t *init_map(size_t x, size_t y);
```
Comment (Sinan-Karakaya) [inline zappy_server/include/map.h:45]: need doc

File: zappy_server/src/free/free_zappy.c [hunk 2]
```diff
@@ -5,29 +5,59 @@
 ** free_all
 */
 
-#include "zappy_server.h"
+#include "free.h"
 #include <stdlib.h>
 
-static size_t free_teams(my_zappy_t *zappy)
+// static void free_client_list(client_list_t *list)
```
Comment (Sinan-Karakaya) [inline zappy_server/src/free/free_zappy.c:11]: Maybe remove commented functions

**Review by Sinan-Karakaya — CHANGES_REQUESTED** [review]
PLease fix my comments

File: zappy_server/src/free/free_zappy.c (hunk 2 above)
Comment (thomasjuin1) [reply zappy_server/src/free/free_zappy.c:11]: In progress

**Review by RoheeAxel — APPROVED** [review]
Nice pr

File: zappy_server/include/map.h (hunk 1 above)
Comment (thomasjuin1) [reply zappy_server/include/map.h:45]: done

**Review by thomasjuin1 — APPROVED** [review]
Code is fine, free in progress

**Review by Sinan-Karakaya — APPROVED** [review]
Looks good

**Pull request merged**